packages/slider/src/ux-slider.ts

```diff
@@ -130,7 +133,7 @@ export class UxSlider implements UxComponent {
         this.updateValue((e as MouseEvent).clientX);
       }
       if (isTouchEvent) {
-        const touches = e.touches as Array<{clientX: number}>;
+        const touches = e.touches as TouchList;
```

Comment:
it is required here because `touches` can also be undefined due to the `?` on line 12.

Reply:
If touches is undefined, then this line will not be executed due to the `if (isTouchEvent)`

Reply:
It's right but typescript doesn't catch this. So I inform TS with the `as ...`

If I didn't want to use the `as ...` I should do multiple `if` checks instead of saving the check in the `const` above. However perf wise I think it's better to save the check and inform typescript what it can't guess because the `as...` will be dropped at compilation time but we reduce the amount of `if` in final code...

would you suggest another way to handle this situation ?

Reply:
Okay, now I understand.

However, in this case I would suggest to use `(e as TouchEvent).touches`, but I'm also fine with this version

Reply:
I understand... however again in this case I would need to rewrite a little more of the method as TS won't let me pretend it could be a `TouchEvent` here. I would either need to go through `as unknown as TouchEvent` or rewrite the method with `e: TouchEvent | MouseEvent` everywhere I use the custom `MouseOrTouchEvent` type.

Again, pros and cons are so minimal here that I prefer to leave it as it is, even though your comments make sense and code could be written differently.

As a side note, the `if` test works in the `instanceof` scenario. TS lets me check even though it's never supposed to be a `TouchEvent` type. Maybe for this reason we might want to use the duck type check that we had above.

We'll see what @bigopon says.

Reply:
We only call this method when it's either mouse/touch. If it's not mouse I think it's safe to assume it's touch. For that `if (!isMouseEvent)` suffices I think.

Reply:
Or a simple `else` https://github.com/aurelia/ux/pull/256/commits/72282695f6193052a06e59d46596b8b7a6e233f6